atom/browser/api/atom_api_cookies.cc

```diff
@@ -305,7 +305,7 @@ v8::Local<v8::Promise> Cookies::Get(const base::DictionaryValue& filter) {
   base::PostTaskWithTraits(
       FROM_HERE, {BrowserThread::IO},
       base::BindOnce(GetCookiesOnIO, base::RetainedRef(getter), std::move(copy),
-                     promise));
+                     std::move(promise)));
```

Comment:
There is no need moving a `scoped_refptr` from UI thread to IO tread, as the `AddRef` is called on the same thread of the constructor.

And in this case it actually should not be moved because `promise` is still being used after this statement.

Getting back on the crash, I think it is caused by `base::Callback` doing copy of bind state, which implicitly copied the `scoped_refptr` passed to it. Changing the `base::Bind` in `GetCookiesOnIO` to `base::BindOnce` should be able to fix the crash, because the latter would move instead of copying its bind state.

The best way fixing it should be making `util::Promise` a move-only type, I have created https://github.com/electron/electron/issues/16799 so anyone wants to work on it can assign themselves.